src/XComponent.ts

```diff
@@ -0,0 +1,69 @@
+import { WebSocketConnection } from "./communication/WebSocketConnection";
+import { Session } from "./interfaces/Session";
+import { Connection } from "./interfaces/Connection";
+import { ErrorListener } from "./interfaces/ErrorListener";
+import { CompositionModel } from "./communication/xcomponentMessages";
+import { Logger, LoggerConfig } from "log4ts";
+import BasicLayout from "log4ts/build/layouts/BasicLayout";
+import ConsoleAppender from "log4ts/build/appenders/ConsoleAppender";
+import { LogLevel } from "log4ts/build/LogLevel";
+import { HeartbeatManager } from "./communication/HeartbeatManager";
+
+export class XComponent {
+    private static logger: Logger = Logger.getLogger("XComponent");
+    private static loggerconfig: LoggerConfig;
+    private static initialized: boolean = false;
+
+    public static connect(serverUrl: string, errorListener?: ErrorListener, heartbeatIntervalSeconds: number = 10): Promise<Connection> {
+        XComponent.tryInitialize();
```

Comment:
can you call the method without referring to the name of the class?
`static` is not a Javascript concept, I'd prefer avoid "typescriptisms" in our interface because it is supposed to be used by JS and TS users, right?

That's the problem I have with the XComponent name too... so you need to call XComponent everywhere on JS code... which is ugly.